### DISCUSSION TITLE
How can I add a directory with Lua module?

### DISCUSSION THREAD

**Discussion by nawordar** [nix-community/nixvim#1328] [Q&A]
In my old pre-Nix Neovim config I already have quite a lot of Lua code split into multiple files and directories. For instance, I have a custom function for formatting which I would like to map to <kbd>&lt;LocalLeader&gt;\\</kbd>. That function calls other functions from a utility module shared with other functions. What would be the best approach to add this directory to my Nixvim config?

**Comment by GaetanLepage** [comment]
You can use the `files` option to output the generated lua code to a different file than `init.lua`.

```nix
files."foo/bar.lua".keymaps = [
  {
    key = "<C-m>";
    action = "foo";
  }
];
```

Will generate a file at `~/.config/nvim/foo/bar.lua` with the following content:
```lua
vim.keymap.set("", "<C-m", "foo", {})
```
(well not exactly this content but an equivalent).

**Reply by nawordar** [reply]
I know about that, but this is not what I want. I have a lot of custom functions in `~/.config/nvim/lua/cdro` in my old config. I would like to move that directory to the new NixVim config and refer to functions defined in it from Nix:
```nix
keymaps = [
  {
    key = "<LocalLeader>\\";
    action = "require('cdro').format()";
    mode = "n";
    lua = true;
  }
];
```
I don't want to embed whole Lua functions in Nix, as their definitions are too long. I think I will end up packaging the directory as a plugin and load it via `extraPlugins`. I just thought there may be a simpler way

**Reply by GaetanLepage** [reply]
Wouldn't the following work ?
```nix
files."lua/crdo/init.lua".extraConfigLua = builtins.readFile ./crdo.lua;
```

Of course, the plugin approach is perfectly fine as well (maybe even cleaner).
